src/distribution.cpp

```diff
@@ -19,32 +19,27 @@
 namespace openmc {
 
 //==============================================================================
-// Discrete implementation
+// DiscreteIndex implementation
 //==============================================================================
 
-Discrete::Discrete(pugi::xml_node node)
+DiscreteIndex::DiscreteIndex(pugi::xml_node node)
 {
   auto params = get_node_array<double>(node, "parameters");
 
-  std::size_t n = params.size();
-  std::vector<double> x_vec(params.begin(), params.begin() + n / 2);
-  std::vector<double> p_vec(params.begin() + n / 2, params.end());
+  prob_.assign(params.begin(), params.end());
```

Comment:
If we are creating a `DiscreteIndex` object from an XML node, chances are it is happening through the `Discrete` class, in which case we really only would want to pick out the last half of the values from `params` rather than the whole thing

Reply:
Or do we even need this method?  If we are doing this through the `Discrete` class, we will have read all the `params` and invoked a constructor for `DiscreteIndex` rather than calling the XML method?

This is only needed if we allow direct creation of `DiscreteIndex` objects via XML that are not part of some other distribution.

Reply:
Went with this method per the comment below...